blobstore/src/main/java/org/jclouds/blobstore/config/LocalBlobStore.java

```diff
@@ -835,22 +836,37 @@ public String completeMultipartUpload(MultipartUpload mpu, List<MultipartPart> p
       ImmutableList.Builder<Blob> blobs = ImmutableList.builder();
       long contentLength = 0;
       Hasher md5Hasher = Hashing.md5().newHasher();
+      MultiBlobInputStream payload;
+      boolean computeMultiPartEtag = false;
+      HashCode mpuHash;
+      String mpuETag;
 
       for (MultipartPart part : parts) {
          Blob blobPart = getBlob(mpu.containerName(), MULTIPART_PREFIX + mpu.id() + "-" + mpu.blobName() + "-" + part.partNumber());
          contentLength += blobPart.getMetadata().getContentMetadata().getContentLength();
          blobs.add(blobPart);
-         md5Hasher.putBytes(BaseEncoding.base16().lowerCase().decode(blobPart.getMetadata().getETag()));
+         if (blobPart.getMetadata().getETag() != null) {
+            md5Hasher.putBytes(BaseEncoding.base16().lowerCase().decode(blobPart.getMetadata().getETag()));
```

Comment:
This seems reasonable, but did you consider returning an empty ETag instead?  I appreciate the commitment to S3 fidelity buy `FilesystemStorageStrategyImpl.getBlob` already returns null ETags instead of recalculating them.  Systems without extended attributes are uncommon so is the additional complexity worth it?

Reply:
I did consider that and thought that previously the blob store returned the hash of the entire content on PUT. Looking back, it looks like it actually returned the hash of an empty string (as the blob part ETag didn't exist). I'll change this patch to the same behavior, keeping the `-{number of parts}` suffix for consistency.

Reply:
You might also want to look at `listMultipartUpload`.  Recalculating ETags can be expensive if a client calls this in some kind of loop!